torch/distributed/_composable/fsdp/_fsdp_init.py

```diff
@@ -58,8 +58,8 @@ def _init_default_fully_shard_mesh() -> DeviceMesh:
     if not dist.distributed_c10d.is_initialized():
         dist.distributed_c10d.init_process_group()
     default_pg = dist.distributed_c10d._get_default_group()
-    device_type = "cuda" if torch.cuda.is_available() else "cpu"
-    mesh = init_device_mesh(device_type, mesh_shape=(default_pg.size(),))
+    device = torch._C._get_accelerator()
+    mesh = init_device_mesh(device.type, mesh_shape=(default_pg.size(),))
```

Comment:
```suggestion
    device = torch.accelerator.current_accelerator() if torch.accelerator.is_available() else torch.device("cpu")
    mesh = init_device_mesh(device.type, mesh_shape=(default_pg.size(),))
```
I think `torch._C._get_accelerator` will be removed in future, refer to https://github.com/pytorch/pytorch/pull/137493.

Reply:
Yes I see, will make change after your PR merged